### PR TITLE
add Mocha tests and run on AppVeyor

### PR DESCRIPTION
**Pull request by ctaggart**

### CODE REVIEW

**Comment by ctaggart** [comment]
I was able to get the Mocha & Chi bindings and imported, but logged a couple of bugs.

I'l like it if the `dotnet fable yarn-build` would build the app and the tests, but I haven't figured out how to do this yet.
https://github.com/fable-compiler/ts2fable/pull/58

File: src/tests.fs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+let mocha: Mocha.IExports = importDefault "Mocha"
+let assertt: Chai.Chai.Assert = import "assert" "Chai"
+
+mocha.describe.Invoke("my suite", fun _ ->
+    mocha.it.Invoke("my test", fun donee ->
+        assertt.equal(1, 1)
+        donee.Invoke()
+    )|> ignore
+    
+) |> ignore
```
Comment (ctaggart) [inline src/tests.fs]: This test runs, but it is not the prettiest API to use.

File: src/Fable.Import.Mocha.fs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        type [<AllowNullLiteral>] TAP =
+            inherit Base
+
+        type [<AllowNullLiteral>] XUnit =
+            inherit Base
+
+        type [<AllowNullLiteral>] XUnitStatic =
+            [<Emit "new $0($1...)">] abstract Create: runner: IRunner * ?options: obj -> XUnit
+
+// module mocha =
```
Comment (ctaggart) [inline src/Fable.Import.Mocha.fs]: bug here. Empty modules should not be printed.

**Comment by ctaggart** [comment]
I separated things into a separate project, but I'm having issues reusing source files. Is it better to reference the project?

current error:
```
C:\Users\camer\fs\ts2fable [mocha ≡ +3 ~5 -3 !]> mocha test/bin/test.js


  my suite
    √ my test

  keywords
    1) done


  1 passing (13ms)
  1 failing

  1) keywords
       done:
     TypeError: (0 , _String.printf) is not a function
      at escapeWord (test\bin\src\keywords.js:28:51)
      at Context.<anonymous> (test\bin\test.js:28:44)
```

**Comment by ctaggart** [comment]
Cool, I got Mocha/Chai tests running in the build. 
image

**Comment by ctaggart** [comment]
With the mocha-appveyor-reporter, the tests show up on the test tab.
https://ci.appveyor.com/project/ctaggart/ts2fable/build/0.2.0-build.18/tests

image

**Comment by alfonsogarciacaro** [comment]
The `printf` error is probably related to this: https://github.com/fable-compiler/Fable/issues/1216

About running the tests after the build, there are several options. In package.json, you can prefix some scripts with `pre` or `post`, so if you add a `postbuild` script, it will always run after running `yarn build`. The other option and also related to your other comment, is to make a project reference in the tests so Fable will automatically compile the main project when compiling the tests. [This is how it's done in fable-powerpack for example](https://github.com/fable-compiler/fable-powerpack/blob/06593dc9bd3a1cc0a4df915b11cf37d521ed3bd7/package.json#L4-L5).

**Pull request merged**